Gems/Atom/RHI/Vulkan/Code/Source/RHI/Device.cpp

```diff
@@ -329,6 +330,23 @@ namespace AZ
                 RETURN_RESULT_IF_UNSUCCESSFUL(result);
             }
 
+            // Create XR session and XR swapchain if XR system is active
+            RHI::XRRenderingInterface* xrSystem = RHI::RHISystemInterface::Get()->GetXRSystem();
+            if (xrSystem)
+            {
+                RHI::Ptr<XRSessionDescriptor> xrSessionDescriptor = aznew XRSessionDescriptor();
+                // Assuming that we will be using the Graphics queue for presentation.
+                xrSessionDescriptor->m_inputData.m_graphicsBinding.m_queueFamilyIndex =
+                    m_commandQueueContext.GetCommandQueue(RHI::HardwareQueueClass::Graphics).GetQueueDescriptor().m_familyIndex;
+                xrSessionDescriptor->m_inputData.m_graphicsBinding.m_queueIndex =
+                    m_commandQueueContext.GetCommandQueue(RHI::HardwareQueueClass::Graphics).GetQueueDescriptor().m_queueIndex;
+                result = xrSystem->CreateSession(xrSessionDescriptor.get());
```

Comment:
> It is done this way so that the CreateSession api is passed a base pointer which can then be cast to the derived class by the OpenXrVk gem.

There is no difference, with `RHI::Ptr<XRSessionDescriptor>` and then passing `xrSessionDescriptor.get()` to `CreateSession` the type is `XRSessionDescriptor*`. And using `XRSessionDescriptor` and then passing `&xrSessionDescriptor` to `CreateSession` is the same `XRSessionDescriptor*`.